File: agent/tools/jira_search_issues.py
import asyncio
from typing import Any

from ..utils.jira import search_issues


def jira_search_issues(jql: str, max_results: int = 50) -> dict[str, Any]:
    """Search Jira issues using JQL (Jira Query Language).

    Args:
        jql: JQL query string. Examples:
            - "project = PROJ AND status = Open"
            - "assignee = currentUser() AND sprint in openSprints()"
            - "created >= -7d ORDER BY created DESC"
        max_results: Maximum number of results to return (default: 50, max: 100)

    Returns:
        Dictionary containing:
        - issues: List of matching issues
        - total: Total number of matching issues
        - start_at: Starting index for pagination
        - max_results: Number of results returned
        - error: Error message if the request failed
    """
    return asyncio.run(search_issues(jql, max_results))

File: src/tools/jiraSearchIssues.ts
/**
 * jira_search_issues tool
 * Mirrors agent/tools/jira_search_issues.py
 */

import { tool } from "@langchain/core/tools";
import { z } from "zod";
import { searchIssues } from "../utils/jira.js";

export const jiraSearchIssues = tool(
  async ({ jql }) => {
    try {
      const result = await searchIssues(jql);
      return JSON.stringify({ success: true, ...result });
    } catch (err) {
      return JSON.stringify({ error: String(err), status: "error" });
    }
  },
  {
    name: "jira_search_issues",
    description:
      "Search Jira issues using JQL (Jira Query Language). " +
      "Example: 'project = PROJ AND status = \"In Progress\" ORDER BY created DESC'",
    schema: z.object({
      jql: z.string().describe("A valid JQL query string."),
    }),
  },
);
